cmd/juju/model/config.go

```diff
@@ -420,9 +420,12 @@ func (c *configCommand) verifyKnownKeys(client configCommandAPI, keys []string)
 
 	allKeys := keys[:]
 	for _, key := range allKeys {
-		// check if the key exists in the known config
-		// and warn the user if the key is not defined
-		if _, exists := known[key]; !exists {
+		// Check if the key exists in the known config
+		// and warn the user if the key is not defined.
+		// If the provided key is authorized-keys, then
+		// we don't warn the user, because we return an
+		// error when trying to update it anyways.
+		if _, exists := known[key]; !exists && key != envconfig.AuthorizedKeysKey {
```

Comment:
I don't know if it is better, but what about instead doing:
```
// AuthorizedKeys is a valid key, though you aren't allowed to set it.
// This will be denied with a server error, but we don't want to warn
// about an unknown key.
known[envconfig.AuthorizedKeysKey] = ""
```
Rather than checking for each non-existent key.
I'm ok with either. The fact that Known isn't evaluated anywhere else makes it ok to do either [we don't look at any of the known values].

This is a lot safer around someone thinking that you should be able to do something with known later on.

Reply:
yeah, I prefer your way indeed. It'll be clear if some day the `known` config is used elsewhere. Done